metis-core/metis-core-common/src/main/java/eu/europeana/metis/core/workflow/plugins/DepublishPlugin.java

```diff
@@ -48,11 +49,11 @@ public DpsTask prepareDpsTask(String datasetId,
       EcloudBasePluginParameters ecloudBasePluginParameters) {
     Map<String, String> extraParameters = new HashMap<>();
     extraParameters.put("METIS_DATASET_ID", datasetId);
-    extraParameters.put("DATASET_DEPUBLISH",
-        Boolean.toString(getPluginMetadata().isDatasetDepublish()));
-    if (!getPluginMetadata().isDatasetDepublish()) {
+    //Do set the records ids parameter only if record ids depublication enabled and there are record ids
+    if (!getPluginMetadata().isDatasetDepublish() && !CollectionUtils
+        .isEmpty(getPluginMetadata().getRecordIdsToDepublish())) {
       extraParameters.put("RECORD_IDS_TO_DEPUBLISH",
-              String.join(",", getPluginMetadata().getRecordIdsToDepublish()));
+          String.join(",", getPluginMetadata().getRecordIdsToDepublish()));
```

Comment:
So if it is NOT marked as a dataset plugin but there are no records (because the user didn't select anything or no records are known), it might accidentally do a dataset depublish? Seems a bit reckless. Perhaps this should throw an exception in this case?

Reply:
Well I had that a responsibility of validating when the workflow creation is submitted.
Should we throw an exception here too? Wouldn't harm I guess..

Reply:
I think that it would be good to throw an exception in this code as well. If we ever do change the validation code, then at least we will see this issue come up quickly.